genomepy/annotation.py

```diff
@@ -0,0 +1,395 @@
+        #
+        # returns 1 because genome header[1] matches the GTF contig names
+
+        # all headers in the file should be formatted the same. so take the first.
+        header = []
+        with _open(self.genome_file, "r") as fa:
+            for line in fa:
+                if line.startswith(">"):
+                    header = line.strip(">\n").split(" ")
+                    break
+
+        for contig in self.annotation_contigs:
+            if contig in header:
+                matching_contig_index = header.index(contig)
+                return matching_contig_index
+        return -1
+
+    @staticmethod
+    def _is_conformable(matching_contig_index: int):
+        if matching_contig_index == 0:
+            raise ValueError("genome and annotation are conforming.")
+        return matching_contig_index > 0
+
+    def _contig_conversion_dict(self, matching_element: int):
+        """
+        build a conversion dict
+        returns the dict and a list of duplicate contig names (should be empty!)
+        """
+        conversion_table = {}
+        duplicate_contigs = []
+        with _open(self.genome_file, "r") as fa:
+            for line in fa:
+                if line.startswith(">"):
+                    line = line.strip(">\n").split(" ")
+                    if line[matching_element] in conversion_table:
+                        duplicate_contigs.append(line[matching_element])
+                    conversion_table[line[matching_element]] = line[0]
+        return conversion_table, list(set(duplicate_contigs))
+
+    def _conform_gtf(
+        self, conversion_dict: dict, filter_contigs: Optional[bool] = True
+    ):
+        """
+        Create an annotation gtf file with contigs conforming to the genome
+        Overwrites the existing gtf file.
+        Returns a list of contigs not present in the genome
+        """
+        missing_contigs = []
+        tmp_dir = mkdtemp(dir=self.genome_dir)
+        new_gtf_file = os.path.join(tmp_dir, os.path.basename(self.annotation_gtf_file))
+        with _open(self.annotation_gtf_file, "r") as old, _open(
+            new_gtf_file, "w"
+        ) as new:
+            for line in old:
+                splitline = line.split("\t")
+                if splitline[0] in conversion_dict:
+                    splitline[0] = conversion_dict[splitline[0]]
+                    line = "\t".join(splitline)
+                    new.write(line)
+                else:
+                    missing_contigs.append(splitline[0])
+                    if not filter_contigs:
+                        new.write(line)
+
+        os.replace(new_gtf_file, self.annotation_gtf_file)
+        rm_rf(tmp_dir)
+        return list(set(missing_contigs))
+
+    def bed_from_gtf(self):
+        """
+        Create an annotation bed file from an annotation gtf file.
+        Overwrites the existing bed file.
+        """
+        tmp_dir = mkdtemp(dir=self.genome_dir)
```

Comment:
do you want to do this the genome_dir? if there is an error, or the program is interrupted, there will be a temp dir there. Why not use the system tempdir?